github_archive/operations/default.json

```diff
@@ -0,0 +1,524 @@
+                      }
+                    },
+                    "aggs": {
+                      "2": {
+                        "cardinality": {
+                          "field": "repo.name"
+                        }
+                      }
+                    }
+                  },
+                  "2": {
+                    "cardinality": {
+                      "field": "repo.name"
+                    }
+                  }
+                }
+              }
+            },
+            "size": 0,
+            "stored_fields": [
+              "*"
+            ],
+            "query": {
+              "bool": {
+                "must": [],
+                "filter": [
+                  {
+                    "range": {
+                      "@timestamp": {
+                        "format": "strict_date_optional_time",
+                        "gte": "2021-09-30T04:00:00.000Z",
+                        "lte": "2021-10-11T04:00:00.000Z"
+                      }
+                    }
+                  }
+                ],
+                "should": [],
+                "must_not": []
+              }
+            }
+          },
+          "request-params": {
+            "wait_for_completion_timeout": "100ms",
+            "keep_on_completion": "true"
+          }
+        }
+      ]
+    },
+    {
+      "stream": [
+        {
+          "name": "aggs-issues-most-reactions",
+          "operation-type": "submit-async-search",
+          "body": {
+            "aggs": {
+              "0": {
+                "terms": {
+                  "field": "payload.issue.html_url",
+                  "order": {
+                    "1": "desc"
+                  },
+                  "size": 10,
+                  "shard_size": 25
+                },
+                "aggs": {
+                  "1": {
+                    "max": {
+                      "field": "payload.issue.reactions.total_count"
+                    }
+                  }
+                }
+              }
+            },
+            "size": 0,
+            "stored_fields": [
+              "*"
+            ],
+            "query": {
+              "bool": {
+                "must": [],
+                "filter": [
+                  {
+                    "range": {
+                      "@timestamp": {
+                        "format": "strict_date_optional_time",
+                        "gte": "2021-09-30T04:00:00.000Z",
+                        "lte": "2021-10-11T04:00:00.000Z"
+                      }
+                    }
+                  }
+                ],
+                "should": [],
+                "must_not": []
+              }
+            }
+          },
+          "request-params": {
+            "wait_for_completion_timeout": "100ms",
+            "keep_on_completion": "true"
+          }
+        }
+      ]
+    },
+    {
+      "operation-type": "get-async-search",
+      "retrieve-results-for": [
```

Comment:
The composite async op was set up following the example at https://esrally.readthedocs.io/en/latest/track.html?highlight=submit-async-search#delete-async-search. My understanding was this composite op was submitting async searches to the cluster, in parallel, then subsequently retrieving results for each async search ID. I've switched to your recommendation.

1s should be enough time to fetch async search results, though more visibility into whether or not we actually received search results would be helpful. It was used since it is the ES default. I'll run some tests to see how async failures are represented.

`simulate-dashboard` is a misleading name for the operation and therefore has been renamed in https://github.com/elastic/rally-tracks/pull/340/commits/96f676515bf32b41cde971c2db455250637e8c40 since it really does not come close to fully simulating dashboard loading.

Reply:
Circling back, https://github.com/elastic/rally-tracks/pull/340/commits/95de2d66d1978b937c6d7969134d1d9a0130f34b adds more work in the `aggs` challenge. `wait_for_completion_timeout` is preserved at `100ms` for async search requests to ensure a `get-async-search` op is required to fetch results.